main/src/com/google/refine/operations/row/RowKeepMatchedOperation.java

```diff
@@ -0,0 +1,81 @@
+
+package com.google.refine.operations.row;
+
+import static com.google.refine.operations.OperationDescription.row_keep_matching_brief;
+import static com.google.refine.operations.OperationDescription.row_keep_matching_rows_desc;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import com.google.refine.browsing.Engine;
+import com.google.refine.browsing.EngineConfig;
+import com.google.refine.browsing.FilteredRows;
+import com.google.refine.browsing.RowVisitor;
+import com.google.refine.history.HistoryEntry;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.model.changes.RowRemovalChange;
+import com.google.refine.operations.EngineDependentOperation;
+
+public class RowKeepMatchedOperation extends EngineDependentOperation {
+
+    @JsonCreator
+    public RowKeepMatchedOperation(
+            @JsonProperty("engineConfig") EngineConfig engineConfig) {
+        super(engineConfig);
+    }
+
+    @Override
+    protected String getBriefDescription(Project project) {
+        return row_keep_matching_brief();
+
+    }
+    protected String createDescription(Project project, int rowCount) {
+        return String.format("%s (%s)",
+                row_keep_matching_brief(),
+                row_keep_matching_rows_desc(rowCount)
```

Comment:
No, you are still concatenating two messages translated independently.
I would like you to use a single message for the entire string.

This means:
* `row_keep_matching_rows_desc=Keep only matching rows ({0} rows removed)`  in the properties file
* `return row_keep_matching_rows_desc(rowCount);` in the Java method.

I am confused as to what has been unclear in my messages above. Could you help me understand how I could have made them clearer, when I wrote "It would be better to have a separate message with a parameter, for the entire string"?